internal/services/groups/groups_data_source.go

```diff
@@ -102,13 +119,15 @@ func groupsDataSourceRead(ctx context.Context, d *schema.ResourceData, meta inte
 			}
 
 			count := len(*result)
-			if count > 1 {
+
+			//Could we make this a switch case?
+			if !mailEnabled && !securityEnabled && count > 1 {
```

Comment:
Assuming we use odata filters, I think this would be better left as `if count > 1`?

Reply:
Yes - Using server side filtering would allow that to return correctly.

If we search for a group with displayname xyz-2 but there is a group xyz-23 does it return both groups or just the one with the exact matching display name? (I assume this is Hamilton and their server doing the clever querying.

Reply:
There are several operators and functions. `eq` is a whole value check. there's also a `startsWith` function which does what you'd expect ([docs](https://docs.microsoft.com/en-us/graph/query-parameters#filter-parameter))